components/core/src/clp/streaming_compression/zstd/Compressor.cpp

```diff
@@ -1,14 +1,51 @@
 #include "Compressor.hpp"
 
-#include "../../Defs.h"
-#include "../../spdlog_with_specializations.hpp"
+#include <cstddef>
+
+#include <spdlog/spdlog.h>
+#include <zstd_errors.h>
+
+#include "../../Array.hpp"
+#include "../../ErrorCode.hpp"
+#include "../../FileWriter.hpp"
+#include "../../TraceableException.hpp"
+#include "../Compressor.hpp"
+#include "../Constants.hpp"
+
+namespace {
+/**
+ * Checks if a value returned by ZStd function indicates an error code.
+ *
+ * For most ZStd functions that return `size_t` results, instead of returning a union type that can
+ * either be a valid result or an error code, an unanimous `size_t` type is returned.
+ * Usually, if the return value exceeds the maximum possible value of valid results, it is treated
+ * as an error code. However, the exact behavior is function-dependent, so ZStd provides:
+ * 1. A value checking function `ZSTD_isError`
+ * 2. A size_t <-> error_code_enum mapping function `ZSTD_getErrorCode`.
+ * See also: https://facebook.github.io/zstd/zstd_manual.html
+ *
+ * @param result A `size_t` type result returned by ZStd functions
+ * @return Whether the result is an error code and indicates an error has occurred
+ */
+auto is_error(size_t result) -> bool {
+    return 0 != ZSTD_isError(result) && ZSTD_error_no_error != ZSTD_getErrorCode(result);
```

Comment:
Do we need both checks to ensure it's an error?
Looking at zstd implementation, I think checking `ZSTD_isError` should be sufficient: https://github.com/facebook/zstd/blob/dev/lib/common/error_private.h#L60
Did u see any doc saying that we should check both?

Reply:
I think checking `ERR_getErrorCode` is sufficient

Reply:
if (!ERR_isError(code)) return (ERR_enum)0; return (ERR_enum) (0-code);

This is what I want

Reply:
I think you misunderstood:
- We shouldn't use `ERR_isError` directly as it's not an exposed API
- I think we should keep this function, but only keep `0 != ZSTD_isError(result)` since this is the recommended way of getting the error code (according to the doc: https://facebook.github.io/zstd/zstd_manual.html)